src/aws_sigv4_internal.erl

```diff
@@ -0,0 +1,261 @@
+%% Based on:
+%% https://github.com/aws/smithy-go/blob/main/aws-http-auth/internal/v4/signer.go
+-module(aws_sigv4_internal).
+
+-export([ do/1
+        , resolve_time/1
+        ]).
+
+%% exported for tests
```

Comment:
How do you feel about wrapping this in an `-ifdef(TEST)`? 👍 

(Also fine with a follow up)

Reply:
I've been burned by `-ifdef(TEST)` in production modules breaking CT-suites. But I'll see if it works for these eunit tests and if so I'll add the ifdef.